realm/realm-library/src/main/java/io/realm/RealmList.java

```diff
@@ -381,6 +381,13 @@ public int size() {
         }
     }
 
+    public boolean contains(E object) {
+        if (!realm.getPath().equals(object.realm.getPath())) {
```

Comment:
And if `object` is a standalone, NPE will be thrown by `object.realm.getPath()`.